src/modules/esl-anchornav/core/esl-anchornav.ts

```diff
@@ -0,0 +1,224 @@
+    if (typeof name !== 'string') return this.setRenderer('default', name);
+    if (typeof name === 'string' && renderer) this._renderers.set(name, renderer);
+  }
+
+  @prop('esl:anchornav:activechanged') public ACTIVECHANGED_EVENT: string;
+  @prop('esl:anchornav:updated') public UPDATED_EVENT: string;
+  @prop('[esl-anchor]') protected ANCHOR_SELECTOR: string;
+
+  /** Item renderer which is used to build inner markup */
+  @attr({defaultValue: 'default', name: 'renderer'}) public rendererName: string;
+
+  protected _active: ESLAnchorData;
+  protected _anchors: ESLAnchorData[] = [];
+  protected _items: Map<string, Element> = new Map();
+  protected _offset: number;
+
+  /** Active anchor */
+  public get active(): ESLAnchorData {
+    return this._active;
+  }
+  public set active(value: ESLAnchorData) {
+    if (this._active === value) return;
+    this._active = value;
+    this._onActiveChange(value);
+  }
+
+  /** Anchors list */
+  protected get $anchors(): HTMLElement[] {
+    return this._anchors.map(({$anchor}) => $anchor);
+  }
+
+  /** Anchornav offset */
+  public get offset(): number {
+    return this._offset || 0;
+  }
+  public set offset(value: number) {
+    if (this._offset === value) return;
+    this._offset = value;
+    memoize.clear(this, '$viewport');
+    this.$$on(this._onAnchorIntersection);
+  }
+
+  /** Anchornav item renderer */
+  protected get itemRenderer(): ESLAnchornavRender | undefined {
+    return ESLAnchornav.getRenderer(this.rendererName) || ESLAnchornav.getRenderer('default');
+  }
+
+  /** Anchornav items container */
+  @memoize()
+  protected get $itemsArea(): HTMLElement {
+    const $provided = this.querySelector<HTMLElement>(`[${this.baseTagName}-items]`);
+    if ($provided) return $provided;
+    const $container = document.createElement('div');
+    $container.setAttribute(this.baseTagName + '-items', '');
+    this.appendChild($container);
+    return $container;
+  }
+
+  /** Anchornav viewport (root element for IntersectionObservers checking visibility) */
+  @memoize()
+  protected get $viewport(): Element | undefined {
+    return getViewportForEl(this);
+  }
+
+  @ready
+  protected override connectedCallback(): void {
+    super.connectedCallback();
+    this.init();
+  }
+
+  /** Initializes the component */
+  public init(): void {
+    this._onAnchornavRequest();
+  }
+
+  /** Updates the component */
+  public update(): void {
+    memoize.clear(this, '$viewport');
+    this.rerender();
+    this.$$on(this._onAnchorIntersection);
+    this.updateActiveAnchor();
+    this._onUpdateEvent();
+  }
+
+  /** Builds the component anchors list markup */
+  protected rerender(): void {
+    const {$itemsArea} = this;
+    const anchors = this.renderAnchors();
+    anchors[0] instanceof Element ? $itemsArea.replaceChildren(...anchors) : $itemsArea.innerHTML = anchors.join('');
+  }
+
+  protected _tempEl: HTMLElement;
+  protected htmlToElement(html: string): Element {
```

Comment:
```suggestion
  // TODO: move to esl-utils helpers
  protected _tempEl: HTMLElement;
  protected htmlToElement(html: string): Element {
```

Reply:
I have modified this method with DOMParser. Looks more correct than inner of div and little bit faster.
Also possible implementation is based on document.implementation.createHTMLDocument.